flocker/node/functional/test_docker.py

```diff
@@ -385,6 +385,28 @@ def got_list(units):
         d.addCallback(got_list)
         return d
 
+    def test_empty_environment(self):
+        """
+        When a container with no custom environment variables is launched via
+        ``DockerClient.add`` the environment in the resulting ``Unit`` returned
+        from ``DockerClient.list`` will ignore the default HOME and PATH
+        environment variables, leaving the ``Unit`` with an Environment of
+        None.
+        """
+        name = random_name()
+        d = self.start_container(name, image_name=u"busybox")
+
+        def started(client):
+            deferred_units = client.list()
+
+            def check_units(units):
+                unit = [unit for unit in units if unit.name == name][0]
+                self.assertIsNone(unit.environment)
+
+            deferred_units.addCallback(check_units)
+        d.addCallback(started)
+        return d
+
     def test_add_with_volumes(self):
```

Comment:
I guess first one is covered by the generic tests below, but second one still needs a test.